EZP-30463 As an Developer, I want to use Solr Cloud (#654)
ezsystems/developer-documentation: docs/guide/solr.md

bin/solr create_core -c collection1 -d server/ez/template
```

##### SolrCloud

> SolrCloud is a cluster of Solr servers. It allows you to:

Reviewer: ```suggestion
SolrCloud is a cluster of Solr servers. It enables you to:
```

- centralize configuration

> - automatically load balancing and fail-over for queries

Reviewer 2: ```suggestion
- automatically load balance and fail-over for queries
```

- integrate ZooKeeper for cluster coordination and configuration

> To set SolrCloud up follow [SolrCloud reference guide.](https://lucene.apache.org/solr/guide/6_6/solrcloud.html)

Reviewer: ```suggestion
To set SolrCloud up follow [SolrCloud reference guide](https://lucene.apache.org/solr/guide/6_6/solrcloud.html).
```

            core: '%solr_core%'
    connections:
        default:

>             distribution_strategy: standalone

Reviewer 3: `standalone` is the default value for the `distribution strategy` so it might be skipped in the example.

                main_translations: endpoint6
```

#### SolrCloud example

> To use SolrCloud you need to specify data distribution strategy for connection via `distribution_strategy` option. The possible values are:

Reviewer: ```suggestion
To use SolrCloud you need to specify data distribution strategy for connection via the `distribution_strategy` option. The possible values are:
```

                main_translations: endpoint6
```

#### SolrCloud example

> To use SolrCloud you need to specify data distribution strategy for connection via the `distribution_strategy` option. The possible values are:

Reviewer 3: ```suggestion
To use SolrCloud you need to specify data distribution strategy for connection via the `distribution_strategy` option to `cloud`
```

- [`cloud`](https://lucene.apache.org/solr/guide/6_6/solrcloud.html) - SolrCloud

The example is based on multi-core setup so any specific language analysis options could be specified on the collection level.

``` yaml
ez_search_engine_solr:
    endpoints:
        main:
            dsn: '%solr_dsn%'
            core: '%solr_main_core%' 
        en:
            dsn: '%solr_dsn%'
            core: '%solr_en_core%'
        fr:
            dsn: '%solr_dsn%'
            core: '%solr_fr_core%'
        # ...
    connections:
        default:
            distribution_strategy: cloud
            entry_endpoints:
                - main
                - en
                - fr
             # -  ...
            mapping:
                translations:
                    eng-GB: en
                    fre-FR: fr
                    # ...
                main_translations: main
```

This solution uses the default SolrCloud [document routing strategy: `compositeId`.](https://lucene.apache.org/solr/guide/6_6/shards-and-indexing-data-in-solrcloud.html#ShardsandIndexingDatainSolrCloud-DocumentRouting)

> In comparison to `implicit` strategy eZ Platform does not need to know shards list.

Reviewer 3: ```suggestion
```
